### ISSUE TITLE
Model training returns Nan as loss value

### ISSUE THREAD

**Issue by jacobeturpin** [jacobeturpin/segment-it#8]
When training, the model returns `Nan` rather than a valid loss number. This could be due to the fact that the model's loss is incorrectly setup for comparison against single-channel binary image mask.

Compiled model
```python
model.compile(optimizer='adam',
              loss=tf.keras.losses.CategoricalCrossentropy(from_logits=True),
              metrics=['accuracy'])
model_history = model.fit(
    train_gen, 
    epochs=EPOCHS,
    steps_per_epoch=TRAIN_STEPS_PER_EPOCH,
    validation_steps=VAL_STEPS_PER_EPOCH,
    validation_data=val_gen,
    #callbacks=[DisplayCallback()])
)
```

Corresponding output
```
Epoch 1/20
 4/26 [===>..........................] - ETA: 9:44 - loss: Nan - accuracy: 0.7641
```
